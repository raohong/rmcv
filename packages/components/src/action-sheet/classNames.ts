import { generateComponentClassNameUtility } from '@rmc-vant/system';
import {
  ActionSheetComponentState,
  ActionSheetNSlot,
  ActionSheetSlot,
} from './interface';

export const ActionSheetName = 'ActionSheet';

export const {
  actionSheetClassNames,
  composeActionSheetSlotClassNames,
  getActionSheetSlotClassNames,
  getActionSheetClassName,
} = generateComponentClassNameUtility<
  typeof ActionSheetName,
  ActionSheetComponentState,
  ActionSheetSlot,
  ActionSheetNSlot
>(
  ActionSheetName,
  {
    root: true,
    title: true,
    description: true,
    content: true,
    cancelButton: true,
    item: true,
    itemSubname: true,
    loadingIcon: true,
    closeIcon: true,
    itemDisabled: true,
    itemLoading: true,
  },
  () => ({
    root: ['root'],
    title: ['title'],
    closeIcon: ['closeIcon'],
    description: ['description'],
    content: ['content'],
    cancelButton: ['cancelButton'],
    item: ['item'],
    itemSubname: ['itemSubname'],
    loadingIcon: ['loadingIcon'],
  }),
);
