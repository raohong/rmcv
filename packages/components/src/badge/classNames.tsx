import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type { BadgeComponentState, BadgeNSlot, BadgeSlot } from './interface';

export const BadgeName = 'Badge';

export const {
  badgeClassNames,
  getBadgeSlotClassNames,
  composeBadgeSlotClassNames,
} = generateComponentClassNameUtility<
  typeof BadgeName,
  BadgeComponentState,
  BadgeSlot,
  BadgeNSlot
>(
  BadgeName,
  {
    root: true,
    number: true,
    numberWrapper: true,
    wrapper: true,
    dot: true,
    positionBottomLeft: true,
    positionBottomRight: true,
    positionTopLeft: true,
    positionTopRight: true,
  },
  ({ dot, position }) => ({
    root: ['root', camelCase(`position-${position}`)],
    wrapper: ['wrapper', dot && 'dot'],
    number: ['number'],
    numberWrapper: ['numberWrapper'],
  }),
);
