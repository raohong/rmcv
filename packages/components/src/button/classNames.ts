import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type { ButtonComponentState, ButtonNSlot, ButtonSlot } from './interface';

export const ButtonName = 'Button';

export const {
  buttonClassNames,
  getButtonSlotClassNames,
  composeButtonSlotClassNames,
} = generateComponentClassNameUtility<
  typeof ButtonName,
  ButtonComponentState,
  ButtonSlot,
  ButtonNSlot
>(
  ButtonName,
  {
    root: true,
    icon: true,
    loading: true,
    disabled: true,
    outlined: true,
    outlinedSizeMini: true,
    outlinedSizeSmall: true,
    outlinedSizeLarge: true,
    outlinedPrimary: true,
    outlinedSuccess: true,
    outlinedWarning: true,
    outlinedDanger: true,
    outlinedShapeRound: true,
    outlinedShapeSquare: true,
    contained: true,
    containedSizeMini: true,
    containedSizeSmall: true,
    containedSizeLarge: true,
    containedPrimary: true,
    containedSuccess: true,
    containedWarning: true,
    containedDanger: true,
    containedShapeRound: true,
    containedShapeSquare: true,
    loadingIcon: true,
    block: true,
  },
  ({ variant, size, shape, loading, disabled, type, block }) => ({
    root: [
      'root',
      variant,
      loading && 'loading',
      disabled && 'disabled',
      size !== 'normal' && camelCase(`${variant}-size-${size}`),
      type !== 'default' && camelCase(`${variant}-${type}`),
      shape !== 'default' && camelCase(`${variant}-shape-${shape}`),
      block && 'block',
    ],
    icon: ['icon'],
    loadingIcon: ['loadingIcon'],
  }),
);
