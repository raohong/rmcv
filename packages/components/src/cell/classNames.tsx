import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type {
  CellComponentState,
  CellGroupComponentState,
  CellGroupNSlot,
  CellGroupSlot,
  CellNSlot,
  CellSlot,
} from './interface';

export const CellName = 'Cell';
export const CellGroupName = 'CellGroup';

export const { cellClassNames, getCellSlotClassNames, composeCellSlotClassNames }
  = generateComponentClassNameUtility<
    typeof CellName,
    CellComponentState,
    CellSlot,
    CellNSlot
  >(
    CellName,
    {
      root: true,
      arrow: true,
      icon: true,
      label: true,
      title: true,
      value: true,
      rightIcon: true,
      sizeLarge: true,
      sizeNormal: true,
    },
    ({ size }) => ({
      root: ['root', camelCase(`size-${size}`)],
      icon: ['icon'],
      title: ['title'],
      value: ['value'],
      label: ['label'],
      rightIcon: ['rightIcon'],
      arrow: ['arrow'],
    }),
  );

export const {
  getCellGroupSlotClassNames,
  composeCellGroupSlotClassNames,
  cellGroupClassNames,
} = generateComponentClassNameUtility<
  typeof CellGroupName,
  CellGroupComponentState,
  CellGroupSlot,
  CellGroupNSlot
>(
  CellGroupName,
  {
    root: true,
    title: true,
    content: true,
    border: true,
    inset: true,
  },
  ({ border, inset }) => ({
    root: ['root', border && 'border', inset && 'inset'],
    title: ['title'],
    content: ['content'],
  }),
);
