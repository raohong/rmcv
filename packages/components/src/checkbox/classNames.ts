import { generateComponentClassNameUtility } from '@rmc-vant/system';
import type {
  CheckboxComponentState,
  CheckboxGroupComponentState,
  CheckboxGroupNSlot,
  CheckboxGroupSlot,
  CheckboxNSlot,
  CheckboxSlot,
} from './interface';

export const CheckboxName = 'Checkbox';
export const CheckboxGroupName = 'CheckboxGroup';

export const {
  checkboxClassNames,
  composeCheckboxSlotClassNames,
  getCheckboxSlotClassNames,
} = generateComponentClassNameUtility<
  typeof CheckboxName,
  CheckboxComponentState,
  CheckboxSlot,
  CheckboxNSlot
>(
  CheckboxName,
  {
    root: true,
    label: true,
    inner: true,
    icon: true,
    checked: true,
    disabled: true,
    input: true,
  },
  ({ checked, disabled }) => ({
    root: ['root', checked && 'checked', disabled && 'disabled'],
    inner: ['inner'],
    icon: ['icon'],
    label: ['label'],
    input: ['input'],
  }),
);

export const {
  checkboxGroupClassNames,
  composeCheckboxGroupSlotClassNames,
  getCheckboxGroupSlotClassNames,
} = generateComponentClassNameUtility<
  typeof CheckboxGroupName,
  CheckboxGroupComponentState,
  CheckboxGroupSlot,
  CheckboxGroupNSlot
>(
  CheckboxGroupName,
  {
    root: true,
    disabled: true,
    horizontal: true,
    vertical: true,
  },
  ({ disabled, direction }) => ({
    root: ['root', disabled && 'disabled', direction],
  }),
);
