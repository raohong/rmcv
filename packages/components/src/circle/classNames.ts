import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type { CircleComponentState, CircleNSlot, CircleSlot } from './interface';

export const CircleName = 'Circle';

export const {
  circleClassNames,
  composeCircleSlotClassNames,
  getCircleSlotClassNames,
} = generateComponentClassNameUtility<
  typeof CircleName,
  CircleComponentState,
  CircleSlot,
  CircleNSlot
>(
  CircleName,
  {
    root: true,
    text: true,
    startPositionBottom: true,
    startPositionLeft: true,
    startPositionRight: true,
    startPositionTop: true,
  },
  ({ startPosition }: CircleComponentState) => ({
    root: ['root', camelCase(`start-position-${startPosition}`)],
    text: ['text'],
  }),
);
