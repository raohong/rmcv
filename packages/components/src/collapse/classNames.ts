import { generateComponentClassNameUtility } from '@rmc-vant/system';
import type { CollapseItemComponentState, CollapseItemSlot } from './interface';

export const CollapseName = 'Collapse';

export const CollapseItemName = 'CollapseItem';

export const {
  collapseItemClassNames,
  getCollapseItemSlotClassNames,
  composeCollapseItemSlotClassNames,
} = generateComponentClassNameUtility<
  typeof CollapseItemName,
  CollapseItemComponentState,
  CollapseItemSlot,
  'root' | 'expandIcon' | 'content'
>(
  CollapseItemName,
  {
    root: true,
    expandIcon: true,
    expanded: true,
    content: true,
    readonly: true,
    disabled: true,
  },
  ({ expanded, disabled, readonly }) => ({
    root: [
      'root',
      expanded && 'expanded',
      disabled && 'disabled',
      readonly && 'readonly',
    ],
    expandIcon: ['expandIcon'],
    content: ['content'],
  }),
);

export const {
  collapseClassNames,
  composeCollapseSlotClassNames,
  getCollapseSlotClassNames,
} = generateComponentClassNameUtility(CollapseName, {
  root: true,
});
