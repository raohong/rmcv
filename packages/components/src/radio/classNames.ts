import { generateComponentClassNameUtility } from '@rmc-vant/system';
import type {
  RadioComponentState,
  RadioGroupComponentState,
  RadioGroupNSlot,
  RadioGroupSlot,
  RadioNSlot,
  RadioSlot,
} from './interface';

export const RadioName = 'Radio';
export const RadioGroupName = 'RadioGroup';

export const {
  radioClassNames,
  composeRadioSlotClassNames,
  getRadioSlotClassNames,
} = generateComponentClassNameUtility<
  typeof RadioName,
  RadioComponentState,
  RadioSlot,
  RadioNSlot
>(
  RadioName,
  {
    root: true,
    label: true,
    inner: true,
    icon: true,
    checked: true,
    disabled: true,
  },
  ({ checked, disabled }) => ({
    root: ['root', checked && 'checked', disabled && 'disabled'],
    inner: ['inner'],
    icon: ['icon'],
    label: ['label'],
  }),
);

export const {
  radioGroupClassNames,
  composeRadioGroupSlotClassNames,
  getRadioGroupSlotClassNames,
} = generateComponentClassNameUtility<
  typeof RadioGroupName,
  RadioGroupComponentState,
  RadioGroupSlot,
  RadioGroupNSlot
>(
  RadioGroupName,
  {
    root: true,
    disabled: true,
    vertical: true,
    horizontal: true,
  },
  ({ disabled, direction }) => ({
    root: ['root', disabled && 'disabled', direction],
  }),
);
