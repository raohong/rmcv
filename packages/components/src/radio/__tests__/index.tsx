import { fireEvent, render, screen } from '@testing-library/react';
import { useState } from 'react';
import { Radio, RadioGroup } from '../';
import { radioClassNames, radioGroupClassNames } from '../classNames';

const testId = 'radio-group';

it('render correctly', () => {
  const tree = render(
    <RadioGroup>
      <Radio value={1}>1</Radio>
      <Radio value={2}> 2</Radio>
    </RadioGroup>,
  );

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with direction', () => {
  render(<RadioGroup direction='horizontal' data-testid={testId}></RadioGroup>);

  expect(screen.getByTestId(testId)).toHaveClass(radioGroupClassNames.horizontal);
});

it('render with defaultValue', () => {
  render(
    <RadioGroup data-testid={testId} defaultValue='1'>
      <Radio data-testid='id1' value='1'>
        name1
      </Radio>
      <Radio value='2'>name2</Radio>
    </RadioGroup>,
  );

  expect(screen.getByTestId('id1')).toHaveClass(radioClassNames.checked);
});

it('render with value', () => {
  const App = () => {
    const [value, setValue] = useState<string>();

    return (
      <RadioGroup onChange={setValue} data-testid={testId} value={value}>
        <Radio data-testid='id1' value='1'>
          name1
        </Radio>
        <Radio data-testid='id2' value='2'>
          name2
        </Radio>
      </RadioGroup>
    );
  };

  render(<App />);

  fireEvent.click(screen.getByTestId('id2'));
  expect(screen.getByTestId('id2')).toHaveClass(radioClassNames.checked);
});
