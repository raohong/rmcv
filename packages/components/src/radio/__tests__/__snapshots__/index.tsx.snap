// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  gap: 8px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: calc(20px - 2px);
  height: 20px;
  min-width: 20px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border: 1px solid #c8c9cc;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  transition-duration: 0.2s;
  transition-property: background-color,color,border-color;
}

.emotion-3 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  opacity: 0;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  cursor: inherit;
}

.emotion-4 {
  line-height: 20px;
  color: #323233;
  box-sizing: border-box;
}

<div
    class="RVuiRadioGroup-root RVuiRadioGroup-vertical emotion-0"
  >
    <label
      class="RVuiRadio-root emotion-1"
    >
      <div
        class="RVuiRadio-inner emotion-2"
      >
        <input
          class="emotion-3"
          type="radio"
          value="1"
        />
      </div>
      <span
        class="RVuiRadio-label emotion-4"
      >
        1
      </span>
    </label>
    <label
      class="RVuiRadio-root emotion-1"
    >
      <div
        class="RVuiRadio-inner emotion-2"
      >
        <input
          class="emotion-3"
          type="radio"
          value="2"
        />
      </div>
      <span
        class="RVuiRadio-label emotion-4"
      >
         2
      </span>
    </label>
  </div>
</DocumentFragment>
`;
